lib/BlockedBy.js

```diff
@@ -49,25 +77,29 @@ class BlockedBy extends NodeResque.Plugin {
      * @method reEnqueue
      * @return {Promise}
      */
-    async reEnqueue() {
+    async reEnqueue(waitingKey, buildId) {
+        // Add the current buildId to the waiting list of this job
+        // Looks like jobID: buildID buildID buildID
+        await this.queueObject.connection.redis.rpush(waitingKey, buildId);
```

Comment:
Everytime `reEnqueue` is invoked, the same build ID is rpush'ed again.

Reply:
Added check for duplicate.